client/my-sites/domains/domain-search/index.tsx

```diff
@@ -435,20 +469,23 @@ export default connect(
 			selectedSiteId: siteId,
 			selectedSiteSlug: getSelectedSiteSlug( state ),
 			domainsWithPlansOnly: currentUserHasFlag( state, DOMAINS_WITH_PLANS_ONLY ),
-			isSiteUpgradeable: isSiteUpgradeable( state, siteId ),
-			isSiteOnMonthlyPlan: isSiteOnMonthlyPlan( state, siteId ),
+			isSiteUpgradeable: !! siteId && isSiteUpgradeable( state, siteId ),
+			isSiteOnMonthlyPlan: !! siteId && isSiteOnMonthlyPlan( state, siteId ),
 			productsList: getProductsList( state ),
 			userCanPurchaseGSuite: canUserPurchaseGSuite( state ),
 			isDomainAndPlanPackageFlow: !! getCurrentQueryArguments( state )?.domainAndPlanPackage,
 			isDomainUpsell:
 				!! getCurrentQueryArguments( state )?.domainAndPlanPackage &&
 				!! getCurrentQueryArguments( state )?.domain,
-			isSiteOnFreePlan: site && isFreePlanProduct( site.plan ),
-			isEcommerceSite:
-				isSiteOnECommerceTrial( state, siteId ) ||
-				isSiteOnWooExpress( state, siteId ) ||
-				isSiteOnEcommerce( state, siteId ),
+			isSiteOnFreePlan: !! site && site.plan && isFreePlanProduct( site.plan ),
```

Comment:
If we convert site to boolean, hence let's do it for `site.plan`, to have consistency:
```suggestion
			isSiteOnFreePlan: !! site && !! site.plan && isFreePlanProduct( site.plan ),
```